lib/restbase.js

```diff
@@ -243,6 +243,10 @@ RESTBase.prototype._request = function(req) {
     var match = priv.router.route(childReq.uri);
     var methods = match && match.value && match.value.methods;
     var handler = methods && (methods[childReq.method] || methods.all);
+    if (childReq.method === 'head' && !handler) {
```

Comment:
I mean we change the shape of an object depending on a condition === polymorphic object, but we can easily avoid that by machining `amagic_bool = null` in coonstructor